src/crewai/knowledge/source/base_knowledge_source.py

```diff
@@ -0,0 +1,39 @@
+from abc import ABC, abstractmethod
+from typing import List
+
+import numpy as np
+from pydantic import BaseModel, ConfigDict, Field
+
+from crewai.knowledge.embedder.base_embedder import BaseEmbedder
+
```

Comment:
extending this to save embeddings to db, then using knowledge class to query from there